### ISSUE TITLE
loadbalancer tests should not assume particular cloud providers do/don't support particular features

### ISSUE THREAD

**Issue by danwinship** [kubernetes/kubernetes#123714]
Various tests in `test/e2e/network/` use `e2skipper.SkipUnlessProviderIs()` (or in one case `SkipIfProviderIs`) to limit the platforms they test on.

There are three problems with this:
- All cloud providers are now out-of-tree, so keeping these correct now requires cross-tree syncing.
- Skipping all "minor" cloud providers by default means that people implementing cloud providers don't get told that they've failed to implement certain functionality, and have no easy way of testing the functionality even if they do implement it.
- The fact that this is necessary at all implies that we have features that may or may not work in any given cluster, where we provide no feedback to the user about whether or not the feature worked.
 
For example, the test "[It should only allow access from service loadbalancer source ranges"](https://github.com/kubernetes/kubernetes/blob/v1.30.0-alpha.3/test/e2e/network/loadbalancer.go#L530) is `SkipUnlessProviderIs("gce", "gke", "aws", "azure")`. Assuming that that skip is actually correct, that implies that every other cloud provider doesn't implement this feature, and users who try to use it to protect their Service will just silently lose.

Ideally, for each of these tests, it should be possible for the test case to figure out, in an entirely platform-independent way, whether the cloud provider supports the feature or not. The obvious way to do this would be to have the cloud provider signal via LoadBalancerStatus. (We obviously can't require providers to signal that they _don't_ implement features they don't know about yet, so this would have to be done in the form of signaling when the provider _did_ implement a particular requested feature.)

(There are also a few non-loadbalancer tests that use `SkipUnlessProviderIs` too: most of these are just GCE-specific tests which will soon never get run in ordinary e2e and probably should be moved out of tree? The other one I see is "It should recreate its iptables rules if they are deleted", which requires a provider in `framework.ProvidersWithSSH`, but that test could be rewritten to use a privileged hostNetwork pod rather than needing ssh...)

(Noticed while adding cloud-provider-kind testing support, https://github.com/kubernetes/test-infra/pull/31659)

/sig network
/sig testing

**Comment by aojea** [comment]
yeah, this is a long standing issue, there should be another issues about this but is worth to repeat it

Nevertheless, the goal now is to completely remove them https://github.com/kubernetes/kubernetes/issues/122828

**Comment by danwinship** [comment]
Closing this in favor of https://github.com/kubernetes/enhancements/issues/4631
